src/linagora.esn.contact/app/addressbook/settings/contact-addressbook-settings.controller.js

```diff
@@ -27,17 +28,25 @@ function contactAddressbookSettingsController(
   self.onCancel = onCancel;
 
   function $onInit() {
-    contactAddressbookService.getAddressbookByBookName($stateParams.bookName, $stateParams.bookId)
-      .then(function(addressbook) {
-        self.addressbook = addressbook;
-        self.addressbookDisplayName = contactAddressbookDisplayService.buildDisplayName(addressbook);
+    if ($stateParams && $stateParams.bookName && $stateParams.bookId) return _fetchAddresBook();
 
-        originalAddressbook = angular.copy(self.addressbook);
+    $rootScope.$on('$stateChangeSuccess', () => {
```

Comment:
do we need to listen to the state change and load the address book each time?

Reply:
It's not "each time". It's only when you enter the url of the settings page. This is necessary or else `$stateParams` will be an empty object because the controller runs before the state has been successfully entered.